Dockerfile

```diff
@@ -1,16 +1,84 @@
-FROM maven:3.5-jdk-8 as builder
```

Comment:
Hi @geminoz! 

There are a number of steps in this `dockerfile` that can increase the image built time. For instance, we are using `ubuntu` image as the base. And from that point onward we have to take care of the installation of every dependency.
IMO, what would be better here is using [Opendk](https://hub.docker.com/_/openjdk/) base image which would have pre-installed dependencies out of the box. We could switch tag for the required `java` version or simple use the latest.

Another thing that could dramatically decreases the built time by reusing the existing targets is writing a dockerfile with a maven plugin. Which would build from existing build files. And can also help ease the development and testing by quickly building an image using existing build files. Something similar i did here [Maven docker plugin](https://github.com/ahab94/iri/blob/8531fb51cc6ff3fe5515a3193b38ae20d0c1b429/pom.xml).

BTW, great stuff i am closing #480 in support of this. 

Please let me know how do you find the proposal.

Thanks